From 8ca085964c8b3f54104a9e08e51b0f6bbac3da47 Mon Sep 17 00:00:00 2001
From: Billy Tsai <billy_tsai@aspeedtech.com>
Date: Wed, 5 Nov 2025 14:52:37 +0800
Subject: [PATCH] pinctrl: aspeed: ast2700_soc1: Fix register offset addresses

Corrects register offsets in pin control driver by adding 0x400 base
offset to all register addresses and changes device mapping to use
parent device instead of current device.
The above change allows the pinctrl driver to access more SCU registers.

Change-Id: Ifda28057abe6bcf968e138b7e59fbc67e4d2ff99
Signed-off-by: Billy Tsai <billy_tsai@aspeedtech.com>
(cherry picked from commit a5181374f9f8b299e77cd8544183e36203da6c30)
Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 drivers/pinctrl/aspeed/pinctrl_ast2700_soc1.c | 148 +++++++++---------
 1 file changed, 74 insertions(+), 74 deletions(-)

diff --git a/drivers/pinctrl/aspeed/pinctrl_ast2700_soc1.c b/drivers/pinctrl/aspeed/pinctrl_ast2700_soc1.c
index 68640264e2ba..4301e6858007 100644
--- a/drivers/pinctrl/aspeed/pinctrl_ast2700_soc1.c
+++ b/drivers/pinctrl/aspeed/pinctrl_ast2700_soc1.c
@@ -37,7 +37,7 @@ static int ast2700_soc1_pinctrl_probe(struct udevice *dev)
 {
 	struct aspeed_pinctrl_priv *priv = dev_get_priv(dev);
 
-	priv->base = dev_remap_addr(dev);
+	priv->base = dev_remap_addr(dev->parent);
 	if (!priv->base)
 		return -ENOMEM;
 
@@ -45,79 +45,79 @@ static int ast2700_soc1_pinctrl_probe(struct udevice *dev)
 }
 
 static struct ast2700_soc1_sig_desc espi1_link[] = {
-	{ 0x58, GENMASK(30, 28) | GENMASK(26, 24) | GENMASK(22, 20) | GENMASK(18, 16) |
+	{ 0x458, GENMASK(30, 28) | GENMASK(26, 24) | GENMASK(22, 20) | GENMASK(18, 16) |
 	GENMASK(14, 12) | GENMASK(10, 8) | GENMASK(6, 4) | GENMASK(2, 0),
 	(2 << 28) | (2 << 24) | (2 << 20) | (2 << 16) | (2 << 12) | (2 << 8) | (2 << 4) | 2},
 };
 
 static struct ast2700_soc1_sig_desc sdio_link[] = {
-	{ 0x00, GENMASK(30, 28) | GENMASK(26, 24) | GENMASK(22, 20) | GENMASK(18, 16) |
+	{ 0x400, GENMASK(30, 28) | GENMASK(26, 24) | GENMASK(22, 20) | GENMASK(18, 16) |
 	GENMASK(14, 12) | GENMASK(10, 8) | GENMASK(6, 4) | GENMASK(2, 0),
 	(3 << 28) | (3 << 24) | (3 << 20) | (3 << 16) | (3 << 12) | (3 << 8) | (3 << 4) | 3},
 };
 
 static struct ast2700_soc1_sig_desc mdio0_link[] = {
-	{ 0x48, GENMASK(22, 16), (1 << 20) | (1 << 16) },
+	{ 0x448, GENMASK(22, 16), (1 << 20) | (1 << 16) },
 };
 
 static struct ast2700_soc1_sig_desc mdio1_link[] = {
-	{ 0x50, GENMASK(22, 16), (1 << 20) | (1 << 16) },
+	{ 0x450, GENMASK(22, 16), (1 << 20) | (1 << 16) },
 };
 
 static struct ast2700_soc1_sig_desc mdio2_link[] = {
-	{ 0x40, GENMASK(6, 0), (1 << 4) | 1 },
+	{ 0x440, GENMASK(6, 0), (1 << 4) | 1 },
 };
 
 static struct ast2700_soc1_sig_desc rgmii0_link[] = {
-	{ 0x44, GENMASK(31, 0),
+	{ 0x444, GENMASK(31, 0),
 	  BIT(0) | BIT(4) | BIT(8) | BIT(12) | BIT(16) | BIT(20) | BIT(24) | BIT(28) },
-	{ 0x48, GENMASK(14, 0), BIT(0) | BIT(4) | BIT(8) | BIT(12) },
+	{ 0x448, GENMASK(14, 0), BIT(0) | BIT(4) | BIT(8) | BIT(12) },
 	/* IO Driving */
-	{ 0xD4, GENMASK(23, 0),
+	{ 0x4D4, GENMASK(23, 0),
 	  BIT(0) | BIT(2) | BIT(4) | BIT(6) | BIT(8) | BIT(10) | BIT(12) | BIT(14) | BIT(16) |
 		  BIT(18) | BIT(20) | BIT(22) }
 };
 
 static struct ast2700_soc1_sig_desc rgmii1_link[] = {
-	{ 0x4c, GENMASK(31, 0),
+	{ 0x44c, GENMASK(31, 0),
 	  BIT(0) | BIT(4) | BIT(8) | BIT(12) | BIT(16) | BIT(20) | BIT(24) | BIT(28) },
-	{ 0x50, GENMASK(14, 0), BIT(0) | BIT(4) | BIT(8) | BIT(12) },
+	{ 0x450, GENMASK(14, 0), BIT(0) | BIT(4) | BIT(8) | BIT(12) },
 	/* IO Driving */
-	{ 0xD8, GENMASK(23, 0),
+	{ 0x4D8, GENMASK(23, 0),
 	  BIT(0) | BIT(2) | BIT(4) | BIT(6) | BIT(8) | BIT(10) | BIT(12) | BIT(14) | BIT(16) |
 		  BIT(18) | BIT(20) | BIT(22) }
 };
 
 static struct ast2700_soc1_sig_desc rmii0_link[] = {
-	{ 0x44,
+	{ 0x444,
 	  GENMASK(30, 28) | GENMASK(22, 20) | GENMASK(18, 16) |
 	  GENMASK(14, 12) | GENMASK(10, 8) | GENMASK(2, 0),
 	  BIT(1) | BIT(9) | BIT(13) | BIT(17) | BIT(21) | BIT(29) },
-	{ 0x48, GENMASK(6, 0), BIT(1) | BIT(5) },
+	{ 0x448, GENMASK(6, 0), BIT(1) | BIT(5) },
 	/* IO Driving */
-	{ 0xD4, GENMASK(23, 0),
+	{ 0x4D4, GENMASK(23, 0),
 	  BIT(0) | BIT(4) | BIT(6) | BIT(8) | BIT(10) | BIT(12) | BIT(14) | BIT(16) |
 		  BIT(18) }
 };
 
 static struct ast2700_soc1_sig_desc rmii0rclko_link[] = {
-	{ 0x44, GENMASK(26, 24), BIT(25) },
+	{ 0x444, GENMASK(26, 24), BIT(25) },
 };
 
 static struct ast2700_soc1_sig_desc rmii1_link[] = {
-	{ 0x4c,
+	{ 0x44c,
 	  GENMASK(30, 28) | GENMASK(22, 20) | GENMASK(18, 16) |
 	  GENMASK(14, 12) | GENMASK(10, 8) | GENMASK(2, 0),
 	  BIT(1) | BIT(9) | BIT(13) | BIT(17) | BIT(21) | BIT(29) },
-	{ 0x50, GENMASK(6, 0), BIT(1) | BIT(5) },
+	{ 0x450, GENMASK(6, 0), BIT(1) | BIT(5) },
 	/* IO Driving */
-	{ 0xD8, GENMASK(23, 0),
+	{ 0x4D8, GENMASK(23, 0),
 	  BIT(0) | BIT(4) | BIT(6) | BIT(8) | BIT(10) | BIT(12) | BIT(14) | BIT(16) |
 		  BIT(18) }
 };
 
 static struct ast2700_soc1_sig_desc rmii1rclko_link[] = {
-	{ 0x4c, GENMASK(26, 24), BIT(25) },
+	{ 0x44c, GENMASK(26, 24), BIT(25) },
 };
 
 static struct ast2700_soc1_sig_desc sgmii_link[] = {
@@ -125,222 +125,222 @@ static struct ast2700_soc1_sig_desc sgmii_link[] = {
 };
 
 static struct ast2700_soc1_sig_desc spi0_link[] = {
-	{ 0x34, GENMASK(2, 0) | GENMASK(6, 4) | GENMASK(10, 8) | GENMASK(14, 12),
+	{ 0x434, GENMASK(2, 0) | GENMASK(6, 4) | GENMASK(10, 8) | GENMASK(14, 12),
 		BIT(0) | BIT(4) | BIT(8) | BIT(12)  },
 };
 
 static struct ast2700_soc1_sig_desc spi0cs1_link[] = {
-	{ 0x34, GENMASK(22, 20), BIT(20) },
+	{ 0x434, GENMASK(22, 20), BIT(20) },
 };
 
 static struct ast2700_soc1_sig_desc spi0abr_link[] = {
-	{ 0x34, GENMASK(26, 24), BIT(24) },
+	{ 0x434, GENMASK(26, 24), BIT(24) },
 };
 
 static struct ast2700_soc1_sig_desc spi0wp_link[] = {
-	{ 0x34, GENMASK(30, 28), BIT(28) },
+	{ 0x434, GENMASK(30, 28), BIT(28) },
 };
 
 static struct ast2700_soc1_sig_desc spi0quad_link[] = {
-	{ 0x34, GENMASK(18, 16) | GENMASK(14, 12), BIT(16) | BIT(12) },
+	{ 0x434, GENMASK(18, 16) | GENMASK(14, 12), BIT(16) | BIT(12) },
 };
 
 static struct ast2700_soc1_sig_desc spi1_link[] = {
-	{ 0x38, GENMASK(2, 0) | GENMASK(6, 4) | GENMASK(10, 8) | GENMASK(14, 12),
+	{ 0x438, GENMASK(2, 0) | GENMASK(6, 4) | GENMASK(10, 8) | GENMASK(14, 12),
 		BIT(0) | BIT(4) | BIT(8) | BIT(12)  },
 };
 
 static struct ast2700_soc1_sig_desc spi1cs1_link[] = {
-	{ 0x38, GENMASK(22, 20), BIT(20) },
+	{ 0x438, GENMASK(22, 20), BIT(20) },
 };
 
 static struct ast2700_soc1_sig_desc spi1abr_link[] = {
-	{ 0x38, GENMASK(26, 24), BIT(24) },
+	{ 0x438, GENMASK(26, 24), BIT(24) },
 };
 
 static struct ast2700_soc1_sig_desc spi1wp_link[] = {
-	{ 0x38, GENMASK(30, 28), BIT(28) },
+	{ 0x438, GENMASK(30, 28), BIT(28) },
 };
 
 static struct ast2700_soc1_sig_desc spi1quad_link[] = {
-	{ 0x38, GENMASK(18, 16) | GENMASK(14, 12), BIT(16) | BIT(12) },
+	{ 0x438, GENMASK(18, 16) | GENMASK(14, 12), BIT(16) | BIT(12) },
 };
 
 static struct ast2700_soc1_sig_desc spi2_link[] = {
-	{ 0x3c, GENMASK(2, 0) | GENMASK(6, 4) | GENMASK(10, 8) | GENMASK(14, 12),
+	{ 0x43c, GENMASK(2, 0) | GENMASK(6, 4) | GENMASK(10, 8) | GENMASK(14, 12),
 		BIT(0) | BIT(4) | BIT(8) | BIT(12)  },
 };
 
 static struct ast2700_soc1_sig_desc spi2cs1_link[] = {
-	{ 0x3c, GENMASK(26, 24), BIT(24) },
+	{ 0x43c, GENMASK(26, 24), BIT(24) },
 };
 
 static struct ast2700_soc1_sig_desc spi2quad_link[] = {
-	{ 0x3c, GENMASK(22, 20) | GENMASK(18, 16), BIT(20) | BIT(16) },
+	{ 0x43c, GENMASK(22, 20) | GENMASK(18, 16), BIT(20) | BIT(16) },
 };
 
 static struct ast2700_soc1_sig_desc fwspiquad_link[] = {
-	{ 0x50, GENMASK(30, 28) | GENMASK(26, 24), BIT(28) | BIT(24) },
+	{ 0x450, GENMASK(30, 28) | GENMASK(26, 24), BIT(28) | BIT(24) },
 };
 
 static struct ast2700_soc1_sig_desc i2c0[] = {
-	{ 0x54, GENMASK(6, 4) | GENMASK(2, 0), BIT(4) | BIT(0) },
+	{ 0x454, GENMASK(6, 4) | GENMASK(2, 0), BIT(4) | BIT(0) },
 };
 
 static struct ast2700_soc1_sig_desc i2c1[] = {
-	{ 0x54, GENMASK(14, 12) | GENMASK(10, 8), BIT(12) | BIT(8) },
+	{ 0x454, GENMASK(14, 12) | GENMASK(10, 8), BIT(12) | BIT(8) },
 };
 
 static struct ast2700_soc1_sig_desc i2c2[] = {
-	{ 0x54, GENMASK(22, 20) | GENMASK(18, 16), BIT(20) | BIT(16) },
+	{ 0x454, GENMASK(22, 20) | GENMASK(18, 16), BIT(20) | BIT(16) },
 };
 
 static struct ast2700_soc1_sig_desc i2c3[] = {
-	{ 0x54, GENMASK(30, 28) | GENMASK(26, 24), BIT(28) | BIT(24) },
+	{ 0x454, GENMASK(30, 28) | GENMASK(26, 24), BIT(28) | BIT(24) },
 };
 
 static struct ast2700_soc1_sig_desc i2c4[] = {
-	{ 0x58, GENMASK(6, 4) | GENMASK(2, 0), BIT(4) | BIT(0) },
+	{ 0x458, GENMASK(6, 4) | GENMASK(2, 0), BIT(4) | BIT(0) },
 };
 
 static struct ast2700_soc1_sig_desc i2c5[] = {
-	{ 0x58, GENMASK(14, 12) | GENMASK(10, 8), BIT(12) | BIT(8) },
+	{ 0x458, GENMASK(14, 12) | GENMASK(10, 8), BIT(12) | BIT(8) },
 };
 
 static struct ast2700_soc1_sig_desc i2c6[] = {
-	{ 0x58, GENMASK(22, 20) | GENMASK(18, 16), BIT(20) | BIT(16) },
+	{ 0x458, GENMASK(22, 20) | GENMASK(18, 16), BIT(20) | BIT(16) },
 };
 
 static struct ast2700_soc1_sig_desc i2c7[] = {
-	{ 0x58, GENMASK(30, 28) | GENMASK(26, 24), BIT(28) | BIT(24) },
+	{ 0x458, GENMASK(30, 28) | GENMASK(26, 24), BIT(28) | BIT(24) },
 };
 
 static struct ast2700_soc1_sig_desc i2c8[] = {
-	{ 0x5c, GENMASK(6, 4) | GENMASK(2, 0), BIT(4) | BIT(0) },
+	{ 0x45c, GENMASK(6, 4) | GENMASK(2, 0), BIT(4) | BIT(0) },
 };
 
 static struct ast2700_soc1_sig_desc i2c9[] = {
-	{ 0x5c, GENMASK(14, 12) | GENMASK(10, 8), BIT(12) | BIT(8) },
+	{ 0x45c, GENMASK(14, 12) | GENMASK(10, 8), BIT(12) | BIT(8) },
 };
 
 static struct ast2700_soc1_sig_desc i2c10[] = {
-	{ 0x5c, GENMASK(22, 20) | GENMASK(18, 16), BIT(20) | BIT(16) },
+	{ 0x45c, GENMASK(22, 20) | GENMASK(18, 16), BIT(20) | BIT(16) },
 };
 
 static struct ast2700_soc1_sig_desc i2c11[] = {
-	{ 0x5c, GENMASK(30, 28) | GENMASK(26, 24), BIT(28) | BIT(24) },
+	{ 0x45c, GENMASK(30, 28) | GENMASK(26, 24), BIT(28) | BIT(24) },
 };
 
 static struct ast2700_soc1_sig_desc i2c12[] = {
-	{ 0x18, GENMASK(14, 12) | GENMASK(6, 4), BIT(14) | BIT(6) },
+	{ 0x418, GENMASK(14, 12) | GENMASK(6, 4), BIT(14) | BIT(6) },
 };
 
 static struct ast2700_soc1_sig_desc i2c13[] = {
-	{ 0x18, GENMASK(22, 20) | GENMASK(18, 16), BIT(22) | BIT(18) },
+	{ 0x418, GENMASK(22, 20) | GENMASK(18, 16), BIT(22) | BIT(18) },
 };
 
 static struct ast2700_soc1_sig_desc i2c14[] = {
-	{ 0x18, GENMASK(30, 28) | GENMASK(26, 24), BIT(30) | BIT(26) },
+	{ 0x418, GENMASK(30, 28) | GENMASK(26, 24), BIT(30) | BIT(26) },
 };
 
 static struct ast2700_soc1_sig_desc i2c15[] = {
-	{ 0x1c, GENMASK(6, 4) | GENMASK(2, 0), BIT(5) | BIT(1) },
+	{ 0x41c, GENMASK(6, 4) | GENMASK(2, 0), BIT(5) | BIT(1) },
 };
 
 static struct ast2700_soc1_sig_desc di2c0[] = {
-	{ 0x0, GENMASK(6, 4) | GENMASK(2, 0), BIT(6) | BIT(2) },
+	{ 0x40, GENMASK(6, 4) | GENMASK(2, 0), BIT(6) | BIT(2) },
 };
 
 static struct ast2700_soc1_sig_desc di2c1[] = {
-	{ 0x0, GENMASK(14, 12) | GENMASK(10, 8), BIT(14) | BIT(10) },
+	{ 0x40, GENMASK(14, 12) | GENMASK(10, 8), BIT(14) | BIT(10) },
 };
 
 static struct ast2700_soc1_sig_desc di2c2[] = {
-	{ 0x0, GENMASK(22, 20) | GENMASK(18, 16), BIT(22) | BIT(18) },
+	{ 0x40, GENMASK(22, 20) | GENMASK(18, 16), BIT(22) | BIT(18) },
 };
 
 static struct ast2700_soc1_sig_desc di2c3[] = {
-	{ 0x0, GENMASK(30, 28) | GENMASK(26, 24), BIT(30) | BIT(26) },
+	{ 0x40, GENMASK(30, 28) | GENMASK(26, 24), BIT(30) | BIT(26) },
 };
 
 static struct ast2700_soc1_sig_desc di2c8[] = {
-	{ 0x2c, GENMASK(6, 4) | GENMASK(2, 0), BIT(5) | BIT(1) },
+	{ 0x42c, GENMASK(6, 4) | GENMASK(2, 0), BIT(5) | BIT(1) },
 };
 
 static struct ast2700_soc1_sig_desc di2c9[] = {
-	{ 0x2c, GENMASK(14, 12) | GENMASK(10, 8), BIT(13) | BIT(9) },
+	{ 0x42c, GENMASK(14, 12) | GENMASK(10, 8), BIT(13) | BIT(9) },
 };
 
 static struct ast2700_soc1_sig_desc di2c10[] = {
-	{ 0x2c, GENMASK(22, 20) | GENMASK(18, 16), BIT(21) | BIT(17) },
+	{ 0x42c, GENMASK(22, 20) | GENMASK(18, 16), BIT(21) | BIT(17) },
 };
 
 static struct ast2700_soc1_sig_desc di2c11[] = {
-	{ 0x2c, GENMASK(30, 28) | GENMASK(26, 24), BIT(29) | BIT(25) },
+	{ 0x42c, GENMASK(30, 28) | GENMASK(26, 24), BIT(29) | BIT(25) },
 };
 
 static struct ast2700_soc1_sig_desc di2c12[] = {
-	{ 0x20, GENMASK(6, 4) | GENMASK(2, 0), BIT(5) | BIT(1) },
+	{ 0x420, GENMASK(6, 4) | GENMASK(2, 0), BIT(5) | BIT(1) },
 };
 
 static struct ast2700_soc1_sig_desc di2c13[] = {
-	{ 0x20, GENMASK(14, 12) | GENMASK(10, 8), BIT(13) | BIT(9) },
+	{ 0x420, GENMASK(14, 12) | GENMASK(10, 8), BIT(13) | BIT(9) },
 };
 
 static struct ast2700_soc1_sig_desc di2c14[] = {
-	{ 0x20, GENMASK(22, 20) | GENMASK(18, 16), BIT(21) | BIT(17) },
+	{ 0x420, GENMASK(22, 20) | GENMASK(18, 16), BIT(21) | BIT(17) },
 };
 
 static struct ast2700_soc1_sig_desc di2c15[] = {
-	{ 0x20, GENMASK(30, 28) | GENMASK(26, 24), BIT(29) | BIT(25) },
+	{ 0x420, GENMASK(30, 28) | GENMASK(26, 24), BIT(29) | BIT(25) },
 };
 
 static struct ast2700_soc1_sig_desc pwm0[] = {
-	{ 0x0c, GENMASK(2, 0), BIT(0) },
+	{ 0x40c, GENMASK(2, 0), BIT(0) },
 };
 
 static struct ast2700_soc1_sig_desc pwm1[] = {
-	{ 0x0c, GENMASK(6, 4), BIT(4) },
+	{ 0x40c, GENMASK(6, 4), BIT(4) },
 };
 
 static struct ast2700_soc1_sig_desc pwm2[] = {
-	{ 0x0c, GENMASK(10, 8), BIT(8) },
+	{ 0x40c, GENMASK(10, 8), BIT(8) },
 };
 
 static struct ast2700_soc1_sig_desc pwm3[] = {
-	{ 0x0c, GENMASK(14, 12), BIT(12) },
+	{ 0x40c, GENMASK(14, 12), BIT(12) },
 };
 
 static struct ast2700_soc1_sig_desc pwm4[] = {
-	{ 0x0c, GENMASK(18, 16), BIT(16) },
+	{ 0x40c, GENMASK(18, 16), BIT(16) },
 };
 
 static struct ast2700_soc1_sig_desc pwm5[] = {
-	{ 0x0c, GENMASK(22, 20), BIT(20) },
+	{ 0x40c, GENMASK(22, 20), BIT(20) },
 };
 
 static struct ast2700_soc1_sig_desc pwm6[] = {
-	{ 0x0c, GENMASK(26, 24), BIT(24) },
+	{ 0x40c, GENMASK(26, 24), BIT(24) },
 };
 
 static struct ast2700_soc1_sig_desc pwm7[] = {
-	{ 0x0c, GENMASK(30, 28), BIT(28) },
+	{ 0x40c, GENMASK(30, 28), BIT(28) },
 };
 
 static struct ast2700_soc1_sig_desc mac0_link[] = {
-	{ 0x68, GENMASK(2, 0), BIT(2) },
+	{ 0x468, GENMASK(2, 0), BIT(2) },
 };
 
 static struct ast2700_soc1_sig_desc mac1_link[] = {
-	{ 0x68, GENMASK(30, 28), BIT(28) | BIT(29) },
+	{ 0x468, GENMASK(30, 28), BIT(28) | BIT(29) },
 };
 
 static struct ast2700_soc1_sig_desc mac2_link[] = {
-	{ 0x68, GENMASK(6, 4), BIT(6) },
+	{ 0x468, GENMASK(6, 4), BIT(6) },
 };
 
 static struct ast2700_soc1_sig_desc pcie2_perst[] = {
-	{ 0x40, GENMASK(2, 0), 2 },
+	{ 0x440, GENMASK(2, 0), 2 },
 };
 
 static const struct aspeed_group_config ast2700_soc1_groups[] = {
-- 
2.34.1

